install.sh

```diff
@@ -1,293 +1,419 @@
+	fi
+else
+	echo "fail"
+	echo "docker-compose is not installed correctly. The most common reason is"
+	echo "having installed docker and docker-compose without using the official"
+	echo "'convenience script'. You may be able to solve this problem by running"
+	if is_python_script "$COMPOSE_CMD_PATH" ; then
+		if is_running_OS_release bookworm ; then
+			echo "   \$ pip3 uninstall -y --break-system-packages docker-compose"
+			echo "   \$ sudo pip3 uninstall -y --break-system-packages docker-compose"
 		else
-      if [ "$NOASKCONFIRM" == "true" ]; then
-        update_docker
-      else
-        if [ ! -f .docker_outofdate ]; then
-          if (whiptail --title "Docker and Docker-Compose Version Issue" --yesno "Docker version is currently $DOCKER_VERSION which is less than $REQ_DOCKER_VERSION consider upgrading or you may experience issues. You will not be prompted again. You can manually upgrade by typing:\n  sudo apt upgrade docker docker-compose\n\nAttempt to upgrade now?" 20 78); then
-            update_docker
-          else
-            touch .docker_outofdate
-          fi
-        fi
-      fi
+			echo "   \$ pip3 uninstall -y docker-compose"
+			echo "   \$ sudo pip3 uninstall -y docker-compose"
 		fi
+		echo "   (ignore any errors from those commands)"
 	else
-		[ -f .docker_outofdate ] && rm .docker_outofdate
-		echo "Docker not installed" >&2
-    if [ "$NOASKCONFIRM" == "true" ]; then
-      do_env_setup
-      install_docker
-    else
-      if [ ! -f .docker_notinstalled ]; then
-        if (whiptail --title "Docker and Docker-Compose" --yesno "Docker is not currently installed, and is required to run IOTstack. Would you like to install docker and docker-compose now?\nYou will not be prompted again." 20 78); then
-            [ -f .docker_notinstalled ] && rm .docker_notinstalled
-            do_env_setup
-            install_docker
-          else
-            touch .docker_notinstalled
-        fi
-      fi
-    fi
+		echo "   \$ sudo apt purge -y docker-compose"
 	fi
-}
-
-function do_env_checks() {
-	GROUPSGOOD=0
+	echo "and then try re-running this script."
+	handle_exit 1
+fi
 
-	if [[ ! "$(user_in_group bluetooth)" == "notgroup" ]] && [[ ! "$(user_in_group bluetooth)" == "true" ]]; then
-	  GROUPSGOOD=1
-    echo "User is NOT in 'bluetooth' group" >&2
-	fi
 
-	if [[ ! "$(user_in_group docker)" == "true" ]]; then
-	  GROUPSGOOD=1
-    echo "User is NOT in 'docker' group" >&2
-	fi
+#----------------------------------------------------------------------
+#							Clone IOTstack repo
+#----------------------------------------------------------------------
 
-	if [ "$GROUPSGOOD" == 1 ]; then
-		echo "!! You might experience issues with docker or bluetooth. To fix run: ./menu.sh --run-env-setup"
+# does the IOTstack folder already exist?
+if [ ! -d "$IOTSTACK" ] ; then
+	# no! clone from GitHub
+	echo -e "\nCloning the IOTstack repository from GitHub"
+	git clone https://github.com/SensorsIot/IOTstack.git "$IOTSTACK"
```

Comment:
Shallow clone better?
```
git clone  --depth=1 https://github.com/SensorsIot/IOTstack.git "$IOTSTACK"
```

Reply:
This will still make fetching expensive, other alternatives are
```
git clone  --filter=tree:0 https://github.com/SensorsIot/IOTstack.git "$IOTSTACK"
```

```
git clone  --filter=blob:none https://github.com/SensorsIot/IOTstack.git "$IOTSTACK"
```

These types of cloning are not intended for developers, but are okay for the average user. Could have an environment variable to disable these for devs.

Reply:
Hmmm.

Well, we're banging into my (lack of) Git knowledge again.

After some Googling, I gather `--depth` refers to commit history, not to the depth of the directory structure. I tinkered with it but the first problem I struck was that a `git switch old-menu` didn't work. So, I think I'll say no to that idea.

Then I tried `--filter=tree:0` which passed my `git switch` test plus a few other tests like applying a PR. It looks like it's doing a lazy fetch. That seems pretty cool.

I'll push another commit against this PR to implement `--filter=tree:0` as an option which can be overridden using `GIT_CLONE_OPTIONS=`. I'll make a matching change in PiBuilder.

<hr>

I didn't try the `blob` variant because, while I *believe* that the only blobs are going to be the image files in the `docs` folder, (a) I'm not 100% sure, (b) I don't know how to verify it, and (c) I have no idea what happens if someone tries to (say) fire up mkdocs and the image files aren't present (a problem that generalises to "what happens if a blob is needed by something other than Git?").

But we can explore that further if you have more info on how it works.

On the flip side, if the objective is just to avoid downloading the docs and images on the assumption that most people will use the web version then wouldn't it be simpler to make the docs folder its own repo? Then people who want to work on the docs can clone it as needed.